### ISSUE TITLE
Reject OCM shares from unkown users

### ISSUE THREAD

**Issue by gmgigi96** [cs3org/reva#3645]
In the context of sciencemesh, ocm should discards the shares received from unknown users (i.e. the ones that didn't accept the invitation workflow)

Original: https://github.com/pondersource/sciencemesh-php/issues/158

**Comment by labkode** [comment]
@gmgigi96 this is true only for the invitation workflow, it should be enforced by configuration. Traditional OCM allows accepting shares from unknown sources.

**Comment by glpatcern** [comment]
Rescoping this: the idea is we want to be able to support "random users" at the pure OCM level, possibly with a warning in the UI (are you sure you want to accept a share from user@unknown.com ?).

Ideally, the configuration should allow the following modes:
* Only ScienceMesh, i.e. reject unknown incoming users and only allow shares towards invited (trusted users
* ScienceMesh + unknown, with warning when accepting incoming shares from unknown users

In the second mode, creating a new OCM share would entail an invitation if the recipient belongs to a ScienceMesh site, or a direct share otherwise.
